test/Snapshot/InMemorySnapshotRepositoryTest.php

```diff
@@ -13,6 +13,31 @@
 class InMemorySnapshotRepositoryTest extends SnapshotRepositoryTest
 {
     /**
+     * @test
+     */
+    public function it_does_not_mutate_state_of_Snapshot_Aggregate()
+    {
+        $aggregate = $this->createAggregateWithHistory(5);
+        $this->repository->save(new Snapshot($aggregate));
+
+        // Applying another event to Snapshotted Aggregate should not affect Snapshot version
+        $aggregate->apply(new MyEvent());
+        $aggregate->getUncommittedEvents();
+
+        $snapshot = $this->repository->load(42);
+        $this->assertEquals(new Snapshot($this->createAggregateWithHistory(5)), $snapshot);
+
+        $loadedAggregate = $snapshot->getAggregateRoot();
```

Comment:
I see what your doing here. I think it helps to extract this to a separate test case to make it easier to understand what the intent is.